lib/dor/services/registration_service.rb

```diff
@@ -4,6 +4,47 @@ module Dor
   class RegistrationService
 
     class << self
+
+      # @TODO: Why isn't all this logic in, for example, Dor::Item.create? or Dor::Base.create? or Dor::Creatable.create?
+      # @TODO: these duplicate checks could be combined into 1 query
+
+      # @param [String] pid an ID to check, if desired.  If not passed (or nil), a new ID is minted
+      # @return [String] a pid you can use immidately, either freshly minted or your checked value
+      # @raise [Dor::DuplicateIdError]
+      def unduplicated_pid(pid = nil)
```

Comment:
Could rename it.  Not an uncommon pattern though, familiar from caching routines.  Don't care about the name much, but don't prefer the new one either.  If you think having the subroutine is bad, all I can say is that it *was* bundled in a single 100+ line method that did **everything** previously...